src/sinks/elasticsearch/common.rs

```diff
@@ -150,10 +150,12 @@ impl ElasticsearchCommon {
                         // to make the transition smoother.
                         Err(error) => {
                             // For now, estimate version.
-                            let assumed_version = match config.suppress_type_name {
-                                Some(true) => 8,
-                                _ => 6,
-                            };
+                            // The `suppress_type_name` option is only valid up to V6, so if a user
+                            // specified that is true, then we will assume they need API V6.
+                            // Otherwise, assume the latest version (V8).
+                            // This is by no means a perfect assumption but it's the best we can
+                            // make with the data we have.
+                            let assumed_version = if config.suppress_type_name { 6 } else { 8 };
```

Comment:
**suggestion (non-blocking)**: would it be useful to log a DEBUG that says we've inferred a version based on the type name option?

Reply:
good to get in the habit of better utilizing debug logging